src/lib/UtapiClient.js

```diff
@@ -256,22 +256,32 @@ export default class UtapiClient {
     * @param {string} reqUid - Request Unique Identifier
     * @param {string} bucket - bucket name
     * @param {number} objectSize - size of object in bytes
+    * @param {number} prevObjectSize - previous size of object in bytes if this
+    * action overwrote an existing object
     * @param {callback} [cb] - (optional) callback to call
     * @return {undefined}
     */
-    pushMetricUploadPart(reqUid, bucket, objectSize, cb) {
+    pushMetricUploadPart(reqUid, bucket, objectSize, prevObjectSize, cb) {
         const callback = cb || this._noop;
         if (this.disableClient) {
             return callback();
         }
         const log = this.log.newRequestLoggerFromSerializedUids(reqUid);
         const timestamp = UtapiClient.getNormalizedTimestamp();
+
+        let oldObjSize = parseInt(prevObjectSize, 10);
+        oldObjSize = isNaN(oldObjSize) ? 0 : oldObjSize;
+        let newObjSize = parseInt(objectSize, 10);
+        newObjSize = isNaN(newObjSize) ? 0 : newObjSize;
```

Comment:
The addition of logs for incorrect value types is a good direction, and one that we are addressing in the https://github.com/scality/utapi/pull/72 refactor of Utapi. With this PR, we are instead enforcing expected types for push metric actions using assertion statements.

`prevObjectSize` can be `null` if the `uploadPart` request is not an overwrite of pre-existing data. In this case, `oldObjSize` should be `0` to create the correct `storageUtilizedDelta`. 

As for `objectSize`, I am following the pattern of `pushMetricPutObject` in which the same situation applies—I assumed there may be a case I do not know of. That value comes from `request.parsedContentLength` in the objectPutPart API.